### PR TITLE
Adds advanced lazarus injectors

### PR DESCRIPTION
**Pull request by aceedex**
A child of the lazarus injector that refreshes mob drops and revives with full HP values. Stocked at 4 in each mining vend and priced at 3000 space bucks.

:cl: 
 *  rscadd: Adds advanced lazarus injectors to the mining vends. These injectors allow full regeneration of mobs they're used on

### CODE REVIEW

**Comment by Probe1** [comment]
On one hand I'm really happy and proud of you for making this.  

On the other I'm pretty sure you can just inherit the majority of what you did with ``` ..() ``` then tack on your revive.

I never knew mobs weren't fully healed.  Interesting.

**Comment by Shadowmech88** [comment]
>I never knew mobs weren't fully healed. Interesting.

They used to be. It was changed in #12182.

File: code/modules/mining/mine_items.dm [hunk 1]
```diff
@@ -709,6 +709,57 @@ proc/move_mining_shuttle()
...
+	icon_state = "adv_lazarus_hypo"
+
+/obj/item/weapon/lazarus_injector/advanced/update_icon()
+	..()
+	if(loaded)
+		icon_state = "adv_lazarus_hypo"
+	else
+		icon_state = "adv_lazarus_empty"
+
+/obj/item/weapon/lazarus_injector/advanced/afterattack(atom/target, mob/user, proximity_flag)
```
Comment (Intigracy) [inline code/modules/mining/mine_items.dm]: just delete this whole proc

on the parent lazarus, define `var/refreshes_drops = FALSE`

change `M.revive(refreshbutcher = 0)` to `M.revive(refreshbutcher = refreshes_drops)` on its afterattack.

Comment (aceedex) [inline code/modules/mining/mine_items.dm]: you mean like this? because if so, it doesn't refresh the mob drops at all
http://pastebin.com/0nztiHDQ

Comment (Intigracy) [reply code/modules/mining/mine_items.dm]: You still have to make the subtype and set that var, just don't override the afterattack.

Comment (aceedex) [inline code/modules/mining/mine_items.dm]: Im not too sure what you mean by that

File: code/modules/mining/mine_items.dm [hunk 2]
```diff
@@ -709,6 +709,57 @@ proc/move_mining_shuttle()
 	if(!loaded)
 		to_chat(user, "<span class='info'>\The [src] is empty.</span>")
 
+/obj/item/weapon/lazarus_injector/advanced
+	name = "Advanced lazarus injector"
```
Comment (PJB3005) [inline code/modules/mining/mine_items.dm]: Remove capitalization.

File: code/modules/mining/mine_items.dm [hunk 3]
```diff
@@ -709,6 +709,57 @@ proc/move_mining_shuttle()
...
+	desc = "A lazarus injector further enhanced with a nanomachine solution. Allows for the complete regeneration of lesser beings"
```
Comment (PJB3005) [inline code/modules/mining/mine_items.dm]: Missed a period here.

File: code/modules/mining/mine_items.dm (hunk 1 above)
Comment (Intigracy) [reply code/modules/mining/mine_items.dm]:
```
/obj/item/weapon/lazarus_injector/advanced
  blablabla the rest of it
  refreshes_drops = TRUE
```

**Comment by Probe1** [comment]
hey hey
ho ho

you're a real coder now

**Comment by Intigracy** [comment]
Exactly what I wanted, thanks.

**Comment by ComicIronic** [comment]
I'm not going to merge this. Some other collab do it.

**Pull request merged**